### DISCUSSION TITLE
Using jaeger-all-in-one for Distributed Tracing and SPM through OTLP

### DISCUSSION THREAD

**Discussion by mapa17** [jaegertracing/jaeger#4580] [Q&A] [answered]
Hello, I want to learn how to use Jaeger for observability in a new project and maybe there is an obvious answer to my question, but I have little understanding of all components yet.

What I want to do is user the jaeger-all-in-one docker image to collect trace information from my apps AND visualize SPM.

Its my understanding that jaeger-all-in-one is the easiest way to collect and visualize traces. I can supposedly us the Tracing API of the Open Telemetry SDK to send data to the jaeger-all-in-one:jaeger-collector.

I am unclear how I can use jaeger-all-in-one for SPM. Reading https://www.jaegertracing.io/docs/1.47/spm/ I get the impression that I need a separate Open Telemetry Collector service that is used by Jaeger Query?

So my question is basically. What do I need to have distributed tracing + SPM? Can i use the jaeger-all-in-one docker image or do I need "more stuff"?

thank you!

**Comment by yurishkuro** [comment]
SPM does not work with just all-in-one, it requires OTEL Collector with an extension that generates the metrics from spans.

**Reply by mapa17** [reply]
Thank you. Do you maybe have a reference to something that I can have a look at?

**Reply by yurishkuro** [reply, marked as answer]
there is an example in https://github.com/jaegertracing/jaeger/blob/main/docker-compose/monitor/
